content/posts/introducing-slight.md

```diff
@@ -0,0 +1,155 @@
+
+SpiderLightning is a cloud system interface that is designed to make applications portable across cloud providers.
+
+It's interface defines a set of common capabilities that are often needed to build distributed applications. These capabilities include key-value stores, message queue, pub/sub, runtime configuration etc.
+
+Application developers can then focus on writing business logic and not worry about the underlying infrastructure.
+
+To experiment with SpiderLightning, we built a host CLI called slight. It embeds a WebAssembly runtime called wasmtime.
+
+## How to use slight?
+
+To use slight, you need to install it first. We assumed that you are using a UNIX-based machine. You can install slight by running the following command:
+
+```bash
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/deislabs/spiderlightning/main/install.sh)"
+```
+
+After installing slight, you can create your very first slight application by running the following command:
+
+```bash
+slight new -n spidey@v0.1.0 rust && cd spidey
+```
+
+This command will create a new rust project called spidey. The project will use the SpiderLightning interface version 0.1.0. We also support C.
+
+Next, you will need to add `wasm32-wasi` target to your rust toolchain. You can do this by running the following command:
+
+```bash
+rustup target add wasm32-wasi
+```
+
+Then you can build the project by running the following command:
+
+```bash
+cargo build --target wasm32-wasi
+```
+
+Finally, you can run the project by running the following command:
+
+```bash
+slight -c slightfile.toml run -m target/wasm32-wasi/debug/spidey.wasm
+```
+
+If you see "Hello, SpiderLightning!", congratulations! You have successfully run your first slight application.
+
+## How does it work?
+
+You might noticed that slight takes two arguments. The first argument is a configuration file called `slightfile.toml`. The second argument is a WebAssembly module. They represent two phases of software development: operation and development, resp.
+
+### Operations
+Let's take a look at the `slightfile.toml` file:
+
+```toml
+specversion = "0.2"
+
+[[capability]]
+resource = "kv.filesystem"
+name = "my-container"
+```
+
+The slight configuration file lists the capabilities that the application needs. In this case, the application needs a key-value store called `my-container`. The key-value store is a filesystem-based key-value store.
+
+Without changing the code, you can change the key-value store to a different infrastructure by changing the configuration file. For example, you can change the key-value store to a Redis-based key-value store by changing the configuration file to the following:
+
+```toml
+specversion = "0.2"
+
+[[capability]]
+resource = "kv.redis"
+name = "my-container"
+    [capability.configs]
+    REDIS_ADDRESS = "redis://127.0.0.1:6379"
+```
```

Comment:
If you want to add capability configs like that, I'd recommend using `slight new -n spidey@v0.2.0 rust`